### DISCUSSION TITLE
Enabling/disabling Xdebug on a per-tasks basis

### DISCUSSION THREAD

**Discussion by AegirLeet** [phpro/grumphp#1102] [Q&A] [answered]
We need Xdebug enabled for PHPUnit code coverage, but it slows down [composer-require-checker](https://github.com/maglnet/ComposerRequireChecker) immensely (see [here](https://github.com/maglnet/ComposerRequireChecker#a-note-about-xdebug)):

```console
$ time XDEBUG_MODE=off ./vendor/bin/composer-require-checker -q

real    0m8.921s
user    0m8.760s
sys     0m0.160s
```

vs.

```console
$ time XDEBUG_MODE=coverage ./vendor/bin/composer-require-checker -q

real    0m49.456s
user    0m49.262s
sys     0m0.190s
```

**Is there any way to enable/disable Xdebug on a per-task basis?**

Patching the task like this does solve the issue, but I'm not sure if that's something that could actually be upstreamed into GrumPHP:

```diff
diff --git a/src/Task/ComposerRequireChecker.php b/src/Task/ComposerRequireChecker.php
index 46ed1ce6..2429a35b 100644
--- a/src/Task/ComposerRequireChecker.php
+++ b/src/Task/ComposerRequireChecker.php
@@ -66,7 +66,7 @@ class ComposerRequireChecker extends AbstractExternalTask

         $process = $this->processBuilder->buildProcess($arguments);

-        $process->run();
+        $process->run(env: ['XDEBUG_MODE' => 'off']);

         if (!$process->isSuccessful()) {
             return TaskResult::createFailed($this, $context, $this->formatter->format($process));
```

Is there a better way?

**Comment by veewee** [comment, marked as answer]
There is currently no way to specify environment variables per task seperately.
Maybe you get better results with pcov instead of xdebug?

**Reply by AegirLeet** [reply]
> Maybe you get better results with pcov instead of xdebug?

I hadn't considered that so far. We set `xdebug.mode = coverage,debug,develop` for development and, ideally, we'd like to keep it that way. So I guess we'd override it with `XDEBUG_MODE=off` in *grumphp.yml* (for all tasks) and use PCOV for coverage. I'll look into it, thanks.

> There is currently no way to specify environment variables per task seperately.

Do you think this would be a useful feature? I could start working on a PR.

**Comment by AegirLeet** [comment]
We set `XDEBUG_MODE=off` and used PCOV instead.
